test/unit/rtc_peer_connection_factory.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import RtcPeerConnectionFactory from '../../src/js/rtc_peer_connection_factory';
+import chai from 'chai';
+import sinon from 'sinon';
+
+describe('RTC Peer Connection Factory', () => {
+    sinon.stub(RtcPeerConnectionFactory.prototype, '_initializeWebSocketEventListeners').returns({});
+    sinon.stub(RtcPeerConnectionFactory.prototype, '_networkConnectivityChecker').returns({});
+    sinon.stub(RtcPeerConnectionFactory.prototype, '_isBrowserSupported').returns(true);
+    var requestAccessStub = sinon.stub().resolves('iceServer');
+    var createPeerConnectionStub = sinon.stub(RtcPeerConnectionFactory.prototype, '_createRtcPeerConnection').returns({});
+    var pcFactory = new RtcPeerConnectionFactory(console, null, null, requestAccessStub, sinon.stub());
+
+    it('check _peerConnectionRequestInFlight eventually resolves to false', async() => {
+        await chai.assert(requestAccessStub.calledOnce);
+        chai.assert(createPeerConnectionStub.calledOnce);
+        chai.assert(createPeerConnectionStub.calledWith('iceServer'));
+        chai.assert.isFalse(pcFactory._peerConnectionRequestInFlight);
+    });
+
+    it('check create peer connection is not called when _peerConnectionRequestInFlight is true', async() => {
+        pcFactory._peerConnectionRequestInFlight = true;
+        pcFactory._requestPeerConnection();
+        chai.assert(requestAccessStub.calledOnce);
```

Comment:
shouldn't be called?

Reply:
I was reusing the same pcFactory object in order to avoid handling two async calls,one in constructor and another in explicit call to _requestPeerConnection(), that make tests complex. However, that made the test asserts to be not intuitive and also made tests dependent on earlier tests. I have updated these tests to reset stub history at the start of each test to make the asserts intuitive and remove dependency on earlier tests.